bindings/python/src/execution.rs

```diff
@@ -0,0 +1,224 @@
+use std::any::Any;
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+
+use arrow_util::pretty;
+use async_trait::async_trait;
+use datafusion::arrow::datatypes::{Schema, SchemaRef};
+use datafusion::arrow::pyarrow::ToPyArrow;
+use datafusion::datasource::TableProvider;
+use datafusion::execution::context::SessionState;
+use datafusion::execution::TaskContext;
+use datafusion::logical_expr::{TableProviderFilterPushDown, TableType};
+use datafusion::physical_plan::stream::RecordBatchStreamAdapter;
+use datafusion::physical_plan::streaming::{PartitionStream, StreamingTableExec};
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion::prelude::Expr;
+use futures::StreamExt;
+use glaredb::{DataFusionError, Operation, RecordBatch, SendableRecordBatchStream};
+use pyo3::exceptions::PyRuntimeError;
+use pyo3::prelude::*;
+use pyo3::types::PyTuple;
+
+use crate::error::PyGlareDbError;
+use crate::runtime::wait_for_future;
+use crate::util::pyprint;
+
+#[pyclass]
+#[derive(Debug, Clone)]
+pub struct PyExecution {
+    op: Arc<Mutex<Operation>>,
```

Comment:
I asked about the rationale for taking a `&mut self` for the Operation methods in the other pr. I think if we make a change to make those take `&self` instead, we can remove the Arc and Mutex here.

Reply:
I removed the mutability and will rebase this in a bit. I still think that we need clone+copy here because of the python runtime constraints, but I'll play with it